portal/src/TextFieldWithImagePreview.tsx

```diff
@@ -0,0 +1,86 @@
+import React, { useCallback, useMemo, useState } from "react";
+import { Label } from "@fluentui/react";
+import FormTextField, { FormTextFieldProps } from "./FormTextField";
+
+import iconPreview from "./images/preview.svg";
+import iconPreviewUnavailable from "./images/preview-unavailable.svg";
+
+import cn from "classnames";
+import { useDebouncedEffect } from "./hook/useDebouncedEffect";
+
+export interface TextFieldWithImagePreviewProps extends FormTextFieldProps {}
+
+const TextFieldWithImagePreview: React.VFC<TextFieldWithImagePreviewProps> =
+  function TextFieldWithImagePreview(props: TextFieldWithImagePreviewProps) {
+    const { label, ...rest } = props;
+    const [imgURL, setImgURL] = useState(rest.value ?? "");
+    const [hasError, setHasError] = useState(false);
+
+    useDebouncedEffect(
+      useCallback(() => {
+        setImgURL(rest.value ?? "");
+        setHasError(false);
+      }, [rest.value]),
+      1000
+    );
+
+    const onError = () => {
+      setHasError(true);
+    };
+
+    const renderImg = useMemo(() => {
+      if (hasError || !imgURL) {
+        return (
+          <div
+            className={cn(
+              "h-10",
+              "w-10",
+              "flex",
+              "flex-row",
+              "justify-center",
+              "items-center",
+              "bg-neutral-lighter",
+              "rounded"
+            )}
+          >
+            {hasError ? (
+              <img src={iconPreviewUnavailable} />
+            ) : (
+              <img src={iconPreview} />
+            )}
+          </div>
+        );
+      }
+      return (
+        <img
```

Comment:
I think we might want to set `object-fit` to `contain`, which is same as the `brand-logo` class in auth ui.